src/spring-cloud/azext_spring_cloud/custom.py

```diff
@@ -921,11 +932,40 @@ def _app_deploy(client, resource_group, service, app, name, version, path, runti
         source=user_source_info)
 
     # upload file
+    logger.warning("[2/3] Uploading package to blob")
     file_service = FileService(storage_name, sas_token=sas_token)
     file_service.create_file_from_path(share_name, None, relative_path, path)
-    logger.info("Upload file succeed, start to update deployment.")
+
+    if file_type == "Source" and ~no_wait:
+        def get_log_url():
+            try:
+                log_url = client.deployments.get_log_file_url(
+                    resource_group_name=resource_group,
+                    service_name=service,
+                    app_name=app,
+                    deployment_name=name).url
+                return log_url
+            except CloudError:
+                return None
+
+        def get_logs_loop():
+            old_log_url = get_log_url()
```

Comment:
Looks like the `log_url` will change once. Just wonder, what if this `old_log_url` has changed when you call it first time? Is it possible?

Reply:
Good questions, we have refined the logic to ensure that `old_log_url` will always be the real old value. In the old logic, when we call it first time, `old_log_url` may be the `new value` sometimes.